Library/ViewModels/RewardPledgeViewModel.swift

```diff
@@ -559,9 +574,19 @@ RewardPledgeViewModelOutputs {
       changePaymentMethodEvent.errors()
     )
 
-    self.showAlert = pledgeErrors
-      .map { $0.errorEnvelope.errorMessages.first }
-      .skipNil()
+    self.showAlert = Signal.merge(
+      pledgeErrors
+        .map { error in
+          let shouldDismiss = (error.errorEnvelope.errorMessages.first == nil
+            || error.errorEnvelope.ksrCode == .UnknownCode)
+          return (message: error.errorEnvelope.errorMessages.first ?? Strings.general_error_something_wrong(),
+                  shouldDismiss: shouldDismiss)
```

Comment:
so my reasoning is that if the message is empty or the server error is unknown, would the user still be able to do anything on this page? they may not understand something is wrong or be able to advance to the next screen, right? This way, if something really weird happens, they can just start the flow again. Is this an imaginary scenario?